include/seqan3/alignment/pairwise/align_result_selector.hpp

```diff
@@ -20,12 +20,36 @@
 #include <seqan3/core/algorithm/configuration.hpp>
 #include <seqan3/core/metafunction/basic.hpp>
 #include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/core/metafunction/transformation_trait_or.hpp>
 #include <seqan3/core/type_list.hpp>
+#include <seqan3/range/decorator/gap_decorator.hpp>
 #include <seqan3/std/ranges>
 
 namespace seqan3::detail
 {
 
+/*!\brief A helper class to define the alignment return type.
+ * \tparam first_t  Type of the first sequence.
+ * \tparam second_t Type of the second sequence.
+ * \details
+ * The type uses the gap decorator if RandomAccessRange and SizedRange are met for both input sequences.
+ */
+template <typename first_t, typename second_t>
+struct alignment_type;
+
+//!\overload
+template <typename first_t, typename second_t>
+    requires std::ranges::RandomAccessRange<first_t> &&
+             std::ranges::SizedRange<first_t> &&
+             std::ranges::RandomAccessRange<second_t> &&
+             std::ranges::SizedRange<second_t>
+struct alignment_type<first_t, second_t>
+{
+    //!\brief The alignment type with gap decorator.
+    using type = std::tuple<decltype(gap_decorator{std::declval<first_t &>()}),
```

Comment:
```suggestion
    using type = std::tuple<view_all<first_t &>,
```

Reply:
I don't understand: why should I remove gap_decorator? Then gaps cannot be stored at all in the alignment result...

Reply:
oops, sorry, I meant:

```c++
     using type = std::tuple<gap_decorator<view_all<first_t &>>,
```